### PR TITLE
simplify datastream entry parsing

### PR DESCRIPTION
**Pull request by V-Staykov**

### CODE REVIEW

File: zk/stages/stage_batches.go [hunk 1]
```diff
@@ -231,149 +224,149 @@ LOOP:
...
-					// when the stage is fired up for the first time
-					log.Warn(fmt.Sprintf("[%s] Skipping block %d, already processed", logPrefix, l2Block.L2BlockNumber))
+				// skip if we already have this block
+				if entry.L2BlockNumber < lastBlockHeight+1 {
+					log.Warn(fmt.Sprintf("[%s] Unwinding to block %d", logPrefix, entry.L2BlockNumber))
+					badBlock, err := eriDb.ReadCanonicalHash(entry.L2BlockNumber)
+					if err != nil {
+						return fmt.Errorf("failed to get bad block: %v", err)
+					}
+					u.UnwindTo(entry.L2BlockNumber, badBlock)
```
Comment (mandrigin) [inline zk/stages/stage_batches.go:306]: when we unwind we actually don't reset what we request from the datastream (bookmark), so we are getting issues with "block number non sequential".

Comment (mandrigin) [reply zk/stages/stage_batches.go:306]: see 
image

Comment (V-Staykov) [reply zk/stages/stage_batches.go:306]: I don't think we should unwind in this case. This unwind doesn't restart the DS client. It just continues using the already downloaded blocks.

Comment (V-Staykov) [reply zk/stages/stage_batches.go:306]: The problem with this is upon disconnecting and then reconnecting. Then  we should reset the bookmark and probably do it to the latest block in the channel.

**Review by revitteth — APPROVED** [review]

**Pull request merged**